### ISSUE TITLE
Add const variables in ChaiScript

### ISSUE THREAD

**Issue by lefticus** [ChaiScript/ChaiScript#17]
Currently, in ChaiScript, there is no way to specify a variable as const.

```
const var myvalue = 5;
```

This feature should be added.

**Comment by ivandmitrievsky** [comment]
Are there any plans on implementing this?

**Comment by lefticus** [comment]
@ivandmitrievsky the issue got a bit derailed, so to clarify, you are interested in being able to specify variables as "const" inside of ChaiScript, correct?

**Comment by jeaye** [comment]
+1 I certainly am. const-correctness is good everywhere we can get it.

**Comment by ivandmitrievsky** [comment]
Correct. I know that it's possible from the `C++` side, but it would be a convinient addition to Chaiscript.
Also, I think there is no need in `const var` declaration, `const` is sufficient by itself and easier to type. Example:
``` c++
const a = 0 // const variable
auto b = 1 // regular variable

**Comment by lefticus** [comment]
Const class attributes would be possible if I define a const variable as something that is assignable exactly once.

```
const a;
a = 0; // OK
a = 1; // runtime error, no assignment operator for const object
```

I'm a little torn on what is the best internal implementation of this. I'll have to ponder it, but I'll try to get something prototyped this week.

@jeaye @ivandmitrievsky are either of you working on public projects that you can share? Please post links to them on the discourse page: http://discourse.chaiscript.com/t/project-announcements/28

**Comment by lefticus** [comment]
@leftibot see if you can add `const` local object support to ChaiScript. Keep the impact minimal - we don't want too much runtime overhead. This should not be terribly difficult I don't think
